code/obj/item/clothing/hats.dm

```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
    desc = "A crimson hat with an enjoyable little pom!"
    icon_state = "pomhat_red"
    item_state = "pomhat_red"
+
+// Mime Beret recolours (mime beret is located elsewhere weirdly)
+
+ABSTRACT_TYPE(/obj/item/clothing/head/frenchberet)
+/obj/item/clothing/head/frenchberet
+	name = "\improper French beret"
+	desc = "Much more artistic than your standard beret."
+	icon = 'icons/obj/clothing/item_hats.dmi'
+	wear_image_icon = 'icons/mob/head.dmi'
+	icon_state = "beret_wht"
+	item_state = "beret_wht"
+	w_class = W_CLASS_TINY
+	throwforce = 0
+
+	white
+		name = "white French beret"
+		icon_state = "beret_wht"
+		item_state = "beret_wht"
+
+	yellow
+		name = "yellow French beret"
+		icon_state = "beret_yel"
+		item_state = "beret_yel"
+
+	mint
+		name = "mint French beret"
+		icon_state = "beret_mnt"
+		item_state = "beret_mnt"
+
+	purple
+		name = "purple French beret"
+		icon_state = "beret_prp"
+		item_state = "beret_prp"
+
+	blue
+		name = "blue French beret"
+		icon_state = "beret_blu"
+		item_state = "beret_blu"
+
+	pink
+		name = "pink French beret"
+		icon_state = "beret_pnk"
+		item_state = "beret_pnk"
+
+	strawberry
+		name = "strawberry beret"
+		icon_state = "beret_strawb"
+		item_state = "beret_strawb"
+
+// Costume goggles
+
+ABSTRACT_TYPE(/obj/item/clothing/head/goggles)
+/obj/item/clothing/head/goggles
+	name = "costume goggles"
+	desc = "They don't even fit over your eyes! How cheap."
+	icon = 'icons/obj/clothing/item_hats.dmi'
+	wear_image_icon = 'icons/mob/head.dmi'
+	icon_state = "goggles_red"
+	item_state = "goggles_red"
+	w_class = W_CLASS_TINY
+	throwforce = 0
+
+	red
+		name = "red costume goggles"
+		icon_state = "goggles_red"
+		item_state = "goggles_red"
+
+	purple
+		name = "purple costume goggles"
+		icon_state = "goggles_prp"
+		item_state = "goggles_prp"
+
+	green
+		name = "green costume goggles"
+		icon_state = "goggles_grn"
+		item_state = "goggles_grn"
+
+	blue
+		name = "blue costume goggles"
+		icon_state = "goggles_blu"
+		item_state = "goggles_blu"
+
+	yellow
+		name = "yellow costume goggles"
+		icon_state = "goggles_yel"
+		item_state = "goggles_yel"
+
+// Baseball Caps
+
+ABSTRACT_TYPE(/obj/item/clothing/head/basecap)
+/obj/item/clothing/head/basecap
+	name = "baseball cap"
+	desc = "Wear it normally, or flip it backwards to increase your coolness."
+	uses_multiple_icon_states = 1
+	icon = 'icons/obj/clothing/item_hats.dmi'
+	wear_image_icon = 'icons/mob/head.dmi'
+	var/hatflip = TRUE
+	var/hatcolour = "black"
+
+	New()
+		..()
+		name = "[hatcolour] baseball cap"
+		item_state = "basecap_[hatcolour]"
+
+	attack_self(var/mob/user as mob)
+		if(user.r_hand == src || user.l_hand == src)
```

Comment:
```suggestion
		
```
This is probably redundant but I'm reviewing this on mobile so I can't guarantee that it's unneeded.